### PR TITLE
Models usage docs revisions

### PR DESCRIPTION
**Pull request by tpdorsey**
Starting revisions of Models usage documentation to match API.

This PR covers up to Error Handling.

## Checklist

* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @samuelcolvin

### CODE REVIEW

**Comment by tpdorsey** [comment]
please review

File: docs/usage/models.md [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
+Untrusted data can be passed to a model and, after parsing and validation, Pydantic guarantees that the fields
 of the resultant model instance will conform to the field types defined on the model.
 
 !!! note
-    *pydantic* is primarily a parsing library, **not a validation library**.
+    Pydantic is primarily a parsing library, **not a validation library**.
     Validation is a means to an end: building a model which conforms to the types and constraints provided.
 
-    In other words, *pydantic* guarantees the types and constraints of the output model, not the input data.
+    In other words, Pydanti guarantees the types and constraints of the output model, not the input data.
```
Comment (dmontagu) [inline docs/usage/models.md:14]:
```suggestion
    In other words, Pydantic guarantees the types and constraints of the output model, not the input data.
```

File: docs/usage/models.md [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
 
     This might sound like an esoteric distinction, but it is not. If you're unsure what this means or
     how it might affect your usage you should read the section about [Data Conversion](#data-conversion) below.
 
-    Although validation is not the main purpose of *pydantic*, you **can** use this library for custom [validation](validators.md).
+    Although validation is not the main purpose of Pydantic, you **can** use this library for custom [validation](validators.md).
```
Comment (dmontagu) [inline docs/usage/models.md:19]: This may be a good place to address the fact that we use the word `validator` in many places to refer to something that might more accurately be described as "transforming". @samuelcolvin not sure how you are thinking about this these days..

File: docs/usage/models.md [hunk 3]
```diff
@@ -29,33 +29,38 @@ class User(BaseModel):
...
-default value, and so a type annotation is not required (however note [this](#field-ordering) warning about field
-order when some fields do not have type annotations).
+In this example, `User` is a model with two fields:
+
+* `id`, which is an integer and is required
+* `name`, which is a string and is not required (it has a default value).
+
+The type of `name` is inferred from the
+default value, and so a type annotation is not required (however, please see the warning on
+[Field Ordering](#field-ordering) regarding field order when some fields do not have type annotations).
```
Comment (dmontagu) [inline docs/usage/models.md:39]:
```suggestion
```

I think let's just get rid of this comment now, as it no longer applies (in v2, all fields _must_ have annotations).

File: docs/usage/models.md [hunk 4]
```diff
@@ -29,33 +29,38 @@ class User(BaseModel):
...
-if no `ValidationError` is raised, you know the resulting model instance is valid.
+In this example, `user` is an instance of `User`.
+Initialisation of the object will perform all parsing and validation.
+If no `ValidationError` is raised, you know the resulting model instance is valid.
 
 ```py group="basic-model"
 assert user.id == 123
 assert isinstance(user.id, int)
-# Note that 123.45 was cast to an int and its value is 123
+# Note that '123' was cast to an int and its value is 123
```
Comment (dmontagu) [inline docs/usage/models.md:52]:
```suggestion
# Note that '123' was coerced to an int and its value is 123
```

File: docs/usage/models.md [hunk 5]
```diff
@@ -29,33 +29,38 @@ class User(BaseModel):
...
 ```
 
-More details on the casting in the case of `user_x` can be found in [Data Conversion](#data-conversion).
+More details on casting can be found in [Data Conversion](#data-conversion).
```
Comment (dmontagu) [inline docs/usage/models.md:55]:
```suggestion
More details on pydantic's coercion logic can be found in [Data Conversion](#data-conversion).
```

File: docs/usage/models.md [hunk 6]
```diff
@@ -76,55 +81,34 @@ assert user.id == 321
...
-: Set of names of fields which were set when the model instance was initialised
-
-`model_fields`
-: a dictionary of the model's fields
-
-`__config__`
-: the configuration class for the model, cf. [model config](model_config.md)
-
-## Recursive Models
+- `copy()`: returns a copy (by default, shallow copy) of the model. See
```
Comment (dmontagu) [inline docs/usage/models.md:89]:
```suggestion
- `model_copy()`: returns a copy (by default, shallow copy) of the model. See
```
Not sure if this is right, but there is a `model_copy` method and I assume that's what we want here?

(Also, I do like that you've included them in alphabetical order and would encourage that, I also think we should reorder them in the actual `main.py` file..)

File: docs/usage/models.md [hunk 7]
```diff
@@ -165,16 +149,16 @@ print(m.model_dump())
 
 For self-referencing models, see [postponed annotations](postponed_annotations.md#self-referencing-models).
 
-## "From Attributes" (aka ORM Mode/Arbitrary Class Instances)
+## Arbitrary class instances
 
-Pydantic models can be created from arbitrary class instances to support models that map to ORM objects.
+(Formerly known as "ORM Mode", or building models from attributes.)
```
Comment (dmontagu) [inline docs/usage/models.md:154]:
```suggestion
(Formerly known as "ORM Mode"/`from_orm`)
```

Comment (dmontagu) [inline docs/usage/models.md:154]:
```suggestion
(Formerly known as "ORM Mode"/`from_orm`)
```

File: docs/usage/models.md [hunk 8]
```diff
@@ -165,16 +149,16 @@ print(m.model_dump())
...
-To do this:
+Pydantic models can be created from arbitrary class instances to support models that map to ORM objects.
 
-1. The [Config](model_config.md) property `from_attributes` must be set to `True`.
-2. The special constructor `from_orm` must be used to create the model instance.
+To do this, use the `model_config` property on the model with `from_attributes` set to `True`. See
+[Model Config](model_config.md) and [ConfigDict](../api/config.md) for more information.
 
-The example here uses SQLAlchemy, but the same approach should work for any ORM.
+The example here uses [SQLAlchemy](https://www.sqlalchemy.org/), but the same approach should work for any ORM.
```
Comment (dmontagu) [inline docs/usage/models.md:161]: For what it's worth, I think we should focus less intently on ORM here — I would phrase it as "Pydantic models can also be created from arbitrary class instances by reading the instance attributes corresponding to the model field names. One common application of this functionality is integration with ORMs." or similar; right now it seems like this functionality is specific to ORMs but it's actually much more general, and I think with the recent name changes, imo, it's worth introducing it in that way.

**Comment by tpdorsey** [comment]
please review
